src/editors/components/EditorFooter/index.test.jsx

```diff
@@ -1,104 +1,96 @@
 import React from 'react';
-import { render, screen } from '@testing-library/react';
-import userEvent from '@testing-library/user-event';
-import { mount } from 'enzyme';
-import EditorFooter from './EditorFooter';
-import EditorPageContext from './EditorPageContext';
-import { ActionStates } from './data/constants';
-import EditorPageProvider from './EditorPageProvider';
-import { saveBlock } from './data/api';
+import { shallow } from 'enzyme';
+import * as module from './index';
+import { selectors, thunkActions } from '../../data/redux';
+import { RequestKeys } from '../../data/constants/requests';
+import { saveTextBlock, navigateCallback } from '../../hooks';
 
-const locationTemp = window.location;
-beforeAll(() => {
-  delete window.location;
-  window.location = {
-    assign: jest.fn(),
-  };
-});
-afterAll(() => {
-  window.location = locationTemp;
-});
+jest.mock('../../data/redux', () => ({
+  thunkActions: {
+    app: {
+      saveBlock: jest.fn().mockName('thunkActions.app.saveBlock'),
+    },
+  },
+  selectors: {
+    app: {
+      isInitialized: jest.fn(state => ({ isInitialized: state })),
+      studioEndpointUrl: jest.fn(state => ({ studioEndpointUrl: state })),
+    },
+    requests: {
+      // eslint-disable-next-line no-unused-vars
+      isFailed: jest.fn((state, requestkey) => ({ isFailed: state })),
```

Comment:
should forward the args for further verification that it is called correctly, as this would pass even if the request key was not passed.
```suggestion
      isFailed: jest.fn((state, params) => ({ isFailed: { state, params } })),
```

Reply:
That makes sense, javascript args are always the hardest to grok